controllers/volume/v1alpha2/volume_controller_unit_test.go

```diff
@@ -674,6 +674,51 @@ func unitTestsReconcile() {
 			})
 		})
 
+		When("VM Spec.Volumes has CNS volume with existing CnsNodeVmAttachment for a different PVC", func() {
+
+			BeforeEach(func() {
+				vmVol = *vmVolumeWithPVC1
+				vm.Spec.Volumes = append(vm.Spec.Volumes, vmVol)
+
+				attachment := cnsAttachmentForVMVolume(vm, vmVol)
+				attachment.Status.Attached = true
+				attachment.Status.AttachmentMetadata = map[string]string{
+					volume.AttributeFirstClassDiskUUID: dummyDiskUUID,
+				}
+				initObjects = append(initObjects, attachment)
+			})
+
+			It("returns success", func() {
+				err := reconciler.ReconcileNormal(volCtx)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Expected VM Status.Volumes", func() {
+					attachment := getCNSAttachmentForVolumeName(vm, vmVol.Name)
+					Expect(attachment).ToNot(BeNil())
+					Expect(vm.Status.Volumes).To(HaveLen(1))
+					assertVMVolStatusFromAttachment(vmVol, attachment, vm.Status.Volumes[0])
+				})
+
+				By("change volume claim name", func() {
+					vmVol.PersistentVolumeClaim.ClaimName = "my-new-claim-name"
+					vm.Spec.Volumes = []vmopv1.VirtualMachineVolume{vmVol}
+
+					err := reconciler.ReconcileNormal(volCtx)
+					Expect(err).ToNot(HaveOccurred())
+				})
+
+				By("new CnsNodeVmAttachment is created with new claim name", func() {
+					attachment := getCNSAttachmentForVolumeName(vm, vmVol.Name)
```

Comment:
Should we consider checking if the UID of this attachment has changed? This could help us verify that the attachment was actually recreated with a new claim name (to ensure that it would be reconciled by CNS).

Reply:
I debated doing that but wasn't something we get for free with the fake client so I decided that Attached going from true to false along with assertVMVolStatusFromAttachmen() was good enough. I'll just fake up a UID on the attachment that is created in the `BeforeEach()`